securedrop_export/export.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import os
+import shutil
+import signal
+import subprocess
+import sys
+import tarfile
+import tempfile
+import time
+
+PRINTER_NAME = "sdw-printer"
+PRINTER_WAIT_TIMEOUT = 60
+DEVICE = "/dev/sda1"
+MOUNTPOINT = "/media/usb"
+ENCRYPTED_DEVICE = "encrypted_volume"
+BRLASER_DRIVER = "/usr/share/cups/drv/brlaser.drv"
+BRLASER_PPD = "/usr/share/cups/model/br7030.ppd"
+
+
+class Metadata(object):
+    """
+    Object to parse, validate and store json metadata from the sd-export archive.
+    """
+
+    METADATA_FILE = "metadata.json"
+    SUPPORTED_EXPORT_METHODS = ["disk", "printer", "printer-test"]
+    SUPPORTED_ENCRYPTION_METHODS = ["luks"]
+
+    def __init__(self, archive_path):
+        self.metadata_path = os.path.join(archive_path, self.METADATA_FILE)
+        try:
+            with open(self.metadata_path) as f:
+                json_config = json.loads(f.read())
+                self.export_method = json_config.get("device", None)
+                self.encryption_method = json_config.get("encryption_method", None)
+                self.encryption_key = json_config.get("encryption_key", None)
+        except Exception as e:
+            msg = "Error parsing metadata: "
+            exit_gracefully(msg, conf, e=e)
```

Comment:
When the packaged script to attempt an export, I get the following error when removing the try/except block in the entrypoint script:
```
Traceback (most recent call last):
  File "/opt/venvs/securedrop-export/lib/python3.5/site-packages/securedrop_export/export.py", line 35, in __init__
    with open(self.metadata_path) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp9phuace8/test-1/metadata.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/send-to-usb", line 10, in <module>
    sys.exit(start())
  File "/opt/venvs/securedrop-export/lib/python3.5/site-packages/securedrop_export/entrypoint.py", line 15, in start
    main.__main__(my_sub)
  File "/opt/venvs/securedrop-export/lib/python3.5/site-packages/securedrop_export/main.py", line 10, in __main__
    submission.archive_metadata = export.Metadata(submission.archive_path)
  File "/opt/venvs/securedrop-export/lib/python3.5/site-packages/securedrop_export/export.py", line 42, in __init__
    exit_gracefully(msg, conf, e=e)
NameError: name 'exit_gracefully' is not defined
```
You may need to invoke `SDExport.exit_gracefully()` here

Reply:
fixed.